### PR TITLE
Replying to comments - LEAN-3894

### PR DESCRIPTION
**Pull request by IslamJMomani**
Given the design: https://www.figma.com/design/h8mdcWWaCnFtWe7OGtLq1y/Reply-to-comments?node-id=18923-8987&t=TI8Rlv6pheZdgsiu-1

- Added the ability to reply to a comment.
- Grouped comments and their replies into threads.
- Added confirmation for comment deletion.
- Implemented expand/collapse functionality for threads.

### CODE REVIEW

File: src/components/comments/CommentActions.tsx [hunk 1]
```diff
@@ -69,13 +74,17 @@ export const CommentActions: React.FC<CommentActionsProps> = ({
   const { isOpen, toggleOpen, wrapperRef } = useDropdown()
 
   return (
-    <>
+    <Container>
       {isResolveEnabled && (
         <CommentResolveButton comment={comment} onClick={toggleResolve} />
       )}
       {isActionsEnabled && (
         <DropdownContainer ref={wrapperRef}>
```
Comment (asouqi) [inline src/components/comments/CommentActions.tsx:82]: should the `DropdownList` be smaller similar to the design

**Comment by asouqi** [comment]
@IslamJMomani should we allow the deletion of comments with a list of replies, instead of deleting replies one by one!

File: src/components/comments/ReplyBox.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  ButtonGroup,
+  PrimaryButton,
+  SecondaryButton,
+} from '@manuscripts/style-guide'
+import React, { ChangeEvent, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+interface ReplyBoxProps {
+  insertCommentReply: (target: string, contents: string) => void
+  commentId: string
```
Comment (mnatsheh) [inline src/components/comments/ReplyBox.tsx:23]: Let's call this `commentID`.

**Review by mnatsheh — APPROVED** [review]

**Comment by IslamJMomani** [comment]
> @IslamJMomani should we allow the deletion of comments with a list of replies, instead of deleting replies one by one!

I discussed this before with Panos and Vip. I suggested following what Google Docs does, which is allowing editing/deletion for any comment or reply. However, they insist they don't want to allow deletion or editing of a comment or reply unless it is the last one.

File: src/components/comments/ReplyBox.tsx (hunk 2 above)
Comment (IslamJMomani) [reply src/components/comments/ReplyBox.tsx]: Done

File: src/components/comments/CommentActions.tsx (hunk 1 above)
Comment (IslamJMomani) [reply src/components/comments/CommentActions.tsx:79]: Done

**Review by asouqi — APPROVED** [review]

**Pull request merged**